scripts/gh-predeploy.js

```diff
@@ -0,0 +1,355 @@
+  fs.mkdirSync(dirString, { recursive: true });
+  const outputPath = `${dirString}/index.html`;
+
+  fs.writeFileSync(outputPath, contents, 'utf-8');
+}
+
+/**
+ * Create the README page (in HTML) from the plugin's README.md file.
+ * This includes the same header, nav bar, and footer as the playground page for a
+ * given package.
+ * 
+ * @param {string} pluginDir The directory of the plugin that is currently being prepared.
+ */
+function createReadmePage(pluginDir) {
+  const packageJson = require(resolveApp(`plugins/${pluginDir}/package.json`));
+  const initialContents = fs.readFileSync(`./plugins/${pluginDir}/README.md`).toString();
+
+  const converter = new showdown.Converter();
+  converter.setFlavor('github');
+  const text = initialContents;
+  const html = converter.makeHtml(text);
+
+  let initialPage = `<!DOCTYPE html>
+  <head>
+    <title></title>
+  </head>
+  <body class="root">
+    <main id="main" class="has-tabs">
+    <article class="article-container site-width">
+      <div class="article">
+      ${html}
+      </div>
+    </article>
+    </main>
+  </body>
+ </html>
+  `;
+
+  // Add the same header, nav bar, and footer as we used for the playground.
+  let modifiedContents = injectHeader(initialPage, packageJson);
+  modifiedContents = injectNavBar(modifiedContents, packageJson, pluginDir);
+  modifiedContents = injectFooter(modifiedContents);
+
+  // Make sure the directory exists, then write to it.
+  const dirString = `./gh-pages/plugins/${pluginDir}/`;
+  fs.mkdirSync(dirString, { recursive: true });
+  const outputPath = `${dirString}/README.html`;
+
+  fs.writeFileSync(outputPath, modifiedContents, 'utf-8');
+}
+
+/**
+ * Copy over files needed to deploy this plugin and its test page to 
+ * github pages.
+ * @param {string} pluginDir The directory with the plugin source files.
+ * @returns 
+ */
+function preparePlugin(pluginDir) {
+  console.log(`Preparing ${pluginDir} plugin for deployment.`);
+  createPluginPage(pluginDir);
+  createReadmePage(pluginDir);
+  return gulp.src(
+    [
+      './plugins/' + pluginDir + '/build/test_bundle.js',
+    ],
+    { base: './plugins/', allowEmpty: true })
+    .pipe(gulp.dest('./gh-pages/plugins/'));
+}
+
+/**
+ * Prepare plugins for deployment to gh-pages.
+ *
+ * For each plugin, copy relevant files to the gh-pages directory.
+ * @param {Function} done Completed callback.
+ * @return {Function} Gulp task.
+ */
+function prepareToDeployPlugins(done) {
+  const dir = 'plugins';
+  const folders = fs.readdirSync(dir).filter(function (file) {
+    return fs.statSync(path.join(dir, file)).isDirectory() &&
+      fs.existsSync(path.join(dir, file, 'package.json')) &&
+      // Only prepare plugins with test pages.
+      fs.existsSync(path.join(dir, file, '/test/index.html'));
+  });
+  return gulp.parallel(folders.map(function (folder) {
+    return function preDeployPlugin() {
+      return preparePlugin(folder);
+    };
+  }))(done);
```

Comment:
Can you use an annonymous function expression here? I think that would make the fact that you're mapping each folder to a function that prepares the folder clearer.

```suggestion
  return gulp.parallel(
      folders.map((folder) => (() => return preparePlugin(folder)))
  )(done);
```

Similarly for `prepareToDeployExamples`

Reply:
I tried this and it doesn't signal completion correctly. I think that means there's an issue with the layers of commands here. Rather than sort it out now, I'm going to look at it after I've settled the details of the rest of the prepare scripts. Ideally I can simplify everything.